File: example/p2p_transaction/p2p.go
package main

import (
	"context"
	"encoding/hex"
	"log"
	"time"

	"github.com/the729/go-libra/client"
	"golang.org/x/crypto/ed25519"
)

const (
	defaultServer = "ac.testnet.libra.org:8000"
	waypoint      = "0:a69511cc7e6d609efcf03e64098056bc3c96d383e0adcf752464a111b081b808"
)

func main() {
	c, err := client.New(defaultServer, waypoint)
	if err != nil {
		log.Fatal(err)
	}
	defer c.Close()

	// We need private key fo the sender to sign the transaction
	priKeyBytes, _ := hex.DecodeString("996911072ee011ffa44a1325e0da593ff3b9374e255115f223cbdffb6bfa0bcfba60d1f8edd6923f59cf9125d3ac80e389afa4e2b8d0e4f1183a30a0270fde71")
	priKey := ed25519.PrivateKey(priKeyBytes)

	// Transaction parameters
	senderAddr := client.MustToAddress("42f5745128c05452a0c68272de8042b1")
	recvAddr := client.MustToAddress("5817cd6e6e84c110c43efca22df54172")
	recvAuthKeyPrefix, _ := hex.DecodeString("26c7bfaa8e0f32206f35bf6d44b43c9c")
	amountMicro := uint64(2 * 1000000)
	maxGasAmount := uint64(500000)
	gasUnitPrice := uint64(0)
	expiration := time.Now().Add(1 * time.Minute)

	log.Printf("Get current account sequence of sender...")
	seq, err := c.QueryAccountSequenceNumber(context.TODO(), senderAddr)
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("... is %d", seq)

	rawTxn, err := client.NewRawP2PTransaction(
		senderAddr, recvAddr, recvAuthKeyPrefix, seq,
		amountMicro, maxGasAmount, gasUnitPrice, expiration,
	)
	if err != nil {
		log.Fatal(err)
	}

	log.Printf("Submit transaction...")
	expectedSeq, err := c.SubmitRawTransaction(context.TODO(), rawTxn, priKey)
	if err != nil {
		log.Fatal(err)
	}

	log.Printf("Waiting until transaction is included in ledger...")
	err = c.PollSequenceUntil(context.TODO(), senderAddr, expectedSeq, expiration)
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("done.")
}

File: example/nodejs/p2p.js
// Use following require statement when using npm.
// const { libra } = require("gopherjs-libra");
const { libra } = require("../../gopherjs-libra/gopherjs-libra.js");

const fromHexString = hexString =>
    new Uint8Array(hexString.match(/.{1,2}/g).map(byte => parseInt(byte, 16)));

const toHexString = bytes =>
    bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');

const defaultServer = "http://hk2.wutj.info:38080",
    waypoint = "0:4d4d0feaa9378069f8fcee71980e142273837e108702d8d7f93a8419e2736f3f";

var senderAddr = fromHexString("42f5745128c05452a0c68272de8042b1"),
    priKey = fromHexString("996911072ee011ffa44a1325e0da593ff3b9374e255115f223cbdffb6bfa0bcfba60d1f8edd6923f59cf9125d3ac80e389afa4e2b8d0e4f1183a30a0270fde71"),
    recvAddr = fromHexString("5817cd6e6e84c110c43efca22df54172"),
    recvAuthKeyPrefix = fromHexString("26c7bfaa8e0f32206f35bf6d44b43c9c");

var client = libra.client(defaultServer, waypoint)
client.queryAccountSequenceNumber(senderAddr)
    .then(r => {
        var txn = {
            "senderAddr": senderAddr,
            "recvAddr": recvAddr,
            "recvAuthKeyPrefix": recvAuthKeyPrefix,
            "senderPrivateKey": priKey,
            "senderSeq": r,
            "amountMicro": 2 * 1000000,
            "maxGasAmount": 500000,
            "gasUnitPrice": 0,
            "expirationTimestamp": parseInt(Date.now() / 1000) + 60,
        };
        return client.submitP2PTransaction(txn);
    })
    .then(r => {
        console.log("Polling sequence number until ", r)
        return client.pollSequenceUntil(senderAddr, r, parseInt(Date.now() / 1000) + 60)
    })
    .then(r => {
        console.log("done.")
    })
    .catch(e => {
        console.log("Error:", e)
    })
